datahub-frontend/app/auth/sso/oidc/OidcCallbackLogic.java

```diff
@@ -300,6 +302,35 @@ private CorpUserSnapshot extractUser(CorpuserUrn urn, CommonProfile profile) {
     return corpUserSnapshot;
   }
 
+  public static Collection<String> getGroupNames(CommonProfile profile, Object groupAttribute, String groupsClaimName) {
+      Collection<String> groupNames = Collections.emptyList();
+      try {
+        if (groupAttribute instanceof Collection) {
+          // List of group names
+          groupNames = (Collection<String>) profile.getAttribute(groupsClaimName, Collection.class);
+        } else if (groupAttribute instanceof String) {
+          String groupString = (String) groupAttribute;
+          ObjectMapper objectMapper = new ObjectMapper();
+          try {
+            // Json list of group names
+            groupNames = objectMapper.readValue(groupString, new TypeReference<List<String>>(){});
+          } catch (Exception e) {
+            if (groupString.contains(",")) {
+              // Comma separated list of group names
+              groupNames = Arrays.asList(groupString.split(","));
+            } else {
+              // Single group name
+              groupNames = Collections.singleton(groupString);
```

Comment:
you actually prob don't need this separate case


Arrays.asList(groupString.split(",")) should return a single array already if the string has no comma. 

e.g. 

"mygroupname" will go to -> ["mygroupname"] in any case I think